shotover/src/sources/opensearch.rs

```diff
@@ -0,0 +1,100 @@
+use crate::codec::{opensearch::OpenSearchCodecBuilder, CodecBuilder, Direction};
+use crate::server::TcpCodecListener;
+use crate::sources::{Source, Transport};
+use crate::tls::{TlsAcceptor, TlsAcceptorConfig};
+use crate::transforms::chain::TransformChainBuilder;
+use anyhow::Result;
+use serde::Deserialize;
+use std::sync::Arc;
+use tokio::sync::{watch, Semaphore};
+use tokio::task::JoinHandle;
+use tracing::{error, info};
+
+#[derive(Deserialize, Debug, Clone)]
+pub struct OpenSearchConfig {
+    pub listen_addr: String,
+    pub connection_limit: Option<usize>,
+    pub hard_connection_limit: Option<bool>,
+    pub tls: Option<TlsAcceptorConfig>,
```

Comment:
either add an integration test for both opensearch Sink and Source TLS support or remove TLS support from this PR.